ext/spl/spl_array.stub.php

```diff
@@ -113,44 +93,41 @@ public function __debugInfo() {}
 
 class ArrayIterator implements SeekableIterator, ArrayAccess, Serializable, Countable
 {
-    /** @param array|object $array */
-    public function __construct($array = [], int $flags = 0) {}
+    public function __construct(array|object $array = [], int $flags = 0) {}
 
     /**
-     * @param mixed $index
+     * @param array|string|float|int|bool|null $index
```

Comment:
As in ArrayObject, we should hint these as `string|int`.

Reply:
Oh yes, I missed those!